### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by mreboud**
This PR improves:

- The documentation of the expression parser
- The documentation on how to add an observable to an analysis file #752

### CODE REVIEW

**Review by MJKirk — COMMENTED** [review]
Other than the one part that isn't clear to me, looks good!

File: doc/defining-observables.rst [hunk 1]
```diff
@@ -29,24 +29,26 @@ Basic rules are used to parse the input string
...
+
+     - Observables are referenced by their :code:`eos.QualifiedName` and must be encapsulated by two chevrons: ``<<...>>``.
+       All features of a qualified name are supported, including specifying option.
+       Examples for a reference to an observable are ``<<B_u->lnu::BR>>`` and ``<<B->Kll::dBR/dq2;q=u>>``.
+     - Parameters are also referenced by their :code:`eos.QualifiedName`, but these must be enclosed in square brackets: ``[[...]]``.
+       For example, ``[[mass::mu]]`` is a valid expression.
+     - Finally, expressions can refer to kinematic variables.
+       The name of the variable must be enclosed in braces: ``{...}``.
+       The kinematic variable can already be used by an observable within the expression.
+       If no observable uses the kinematic variable, a new kinematic variable is automatically created.
```
[the remark this answers is not in the document]
Comment (mreboud) [reply doc/defining-observables.rst:42]: Yes, I should remove them. The idea is that when the parser reads a kinematic variable, it checks whether this variable name was already used by an observable. If this is not the case, it internally adds the variable to the list of used variable. The aim is that when you call your new defined observable, EOS should complain if an used variable name is not provided in the kinematic specification.

So this is internal business and should be removed. What do you think @dvandyk?

[the remark this answers is not in the document]
Comment (MJKirk) [reply doc/defining-observables.rst:42]: Okay, so the point is you can define new kinematic variables within the string? E.g. something like
`
<<B->K::f_+(q2)>> * (1 - {my_new_q2})
`
and this is now an observable that requires two kinematic variables to be given? And EOS can give a useful error that names the missing variable if you don't? 

If so, I would keep this information, but reword it and move it to a new bullet point, as it is currently under "Expressions can refer to existing EOS objects"

[the remark this answers is not in the document]
Comment (mreboud) [reply doc/defining-observables.rst:42]: Yes, and I agree, I will do that! Thanks for the feedback :)

[the remark this answers is not in the document]
Comment (dvandyk) [reply doc/defining-observables.rst:42]: > So this is internal business and should be removed. What do you think @dvandyk?

Yes, remove it. And maybe slightly adapt it to this?
```
- Expressions can also explicitly involve kinematic variables.
       To refer to a kinematic variable, its name must be enclosed in braces: ``{...}``.
```

**Comment by MJKirk** [comment]
Just fixed a spelling mistake, but otherwise looks good!

**Review by MJKirk — APPROVED** [review]

**Pull request merged**